doc/release/1.7.0-notes.rst

```diff
@@ -24,53 +24,290 @@ For running on PyPy, PyPy3 6.0+ is required.
+
 `scipy.integrate` improvements
 ------------------------------
 
 
 `scipy.interpolate` improvements
 --------------------------------
 
+Improved input validation and error messages for ``fitpack.bispev`` and
+``fitpack.parder`` for scenarios that previously caused substantial confusion
+for users.
+
 `scipy.io` improvements
 -----------------------
 
 
 `scipy.linalg` improvements
 ---------------------------
 
+An LAPACK wrapper was added for access to the ``tgexc`` subroutine.
+
 
 `scipy.ndimage` improvements
 ----------------------------
 
+`scipy.ndimage.affine_transform` is now able to infer the ``output_shape`` from
+the ``out`` array.
 
 `scipy.optimize` improvements
 -----------------------------
 
+The optional parameter ``bounds`` was added to 
+`scipy.optimize.optimize._minimize_neldermead` to support bounds constraints
+for the Nelder-Mead solver.
+
+``trustregion`` methods ``trust-krylov``, ``dogleg`` and ``trust-ncg`` can now
+estimate ``hess`` by finite difference using one of 
+``["2-point", "3-point", "cs"]``.
+
+``halton`` was added as a ``sampling_method`` in `scipy.optimize.shgo`. 
+``sobol`` was fixed and is now using `scipy.stats.qmc.Sobol`.
+
+``halton`` and ``sobol`` were added as ``init`` methods in 
+`scipy.optimize.differential_evolution.`
+
+``differential_evolution`` now accepts an ``x0`` parameter to provide an
+initial guess for the minimization.
+
+``least_squares`` has a modest performance improvement when SciPy is built
+with Pythran transpiler enabled
 
 `scipy.signal` improvements
 ---------------------------
 
+``get_window`` supports ``general_cosine`` and ``general_hamming`` window 
+functions.
+
+`scipy.signal.medfilt2d` now releases the GIL where appropriate to enable
+performance gains via multithreaded calculations.
 
 `scipy.sparse` improvements
 ---------------------------
 
+Addition of ``dia_matrix`` sparse matrices is now faster.
+
+
 `scipy.spatial` improvements
 ----------------------------
 
+``distance.cdist`` and ``distance.pdist`` performance has greatly improved for 
+certain weighted metrics. Namely: ``minkowski``, ``euclidean``, ``chebyshev``,
+``canberra``, and ``cityblock``.
+
+Modest performance improvements for many of the unweighted ``cdist`` and 
+``pdist`` metrics noted above.
+
+The parameter ``seed`` was added to `scipy.spatial.vq.kmeans` and 
+`scipy.spatial.vq.kmeans2`.
+
+The parameters ``axis`` and ``keepdims`` where added to 
+`scipy.spatial.distance.jensenshannon.`
+
+The ``rotation`` methods `from_rotvec` and `as_rotvec` now accept a `degrees`
+argument to specify usage of degrees instead of radians.
+
+`scipy.special` improvements
+--------------------------
+
+Wright's generalized Bessel function for positive arguments was added as
+`scipy.special.wright_bessel.`
+
+An implementation of the inverse of the Log CDF of the Normal Distribution is
+now available via `scipy.special.ndtri_exp`.
+
 `scipy.stats` improvements
 --------------------------
```

Comment:
I am working these; please avoid pushing changes to this in the meantime.

Reply:
Done with changes. Please feel free to edit.

Also, there are a few PRs we'll want to add here if/when they merge:

- gh-13732 (I plan to merge tomorrow)
- gh-13951 (I don't think there are any code changes needed; just some final checks)
- gh-13319 (if we can reach consensus)

Reply:
Hmm after my changes, it no longer renders when I "View File". Anyone see what went wrong in tylerjereddy/scipy#52?

Reply:
There were several warnings in the `make html-scipyorg` build for example, causing it to fail.

Not sure if that is what you mean, but I have pushed in adjustments to fix all the warnings (at least locally the build succeeds now).

Note that for some of the `stats` "sections" I changed the `^^^` underlines to `---` to fix warnings like this:

```
release/1.7.0-notes.rst:148: WARNING: Title level inconsistent:

Hypothesis Tests
^^^^^^^^^^^^^^^^
release/1.7.0-notes.rst:184: WARNING: Title level inconsistent:

Sample statistics
^^^^^^^^^^^^^^^^^
release/1.7.0-notes.rst:214: WARNING: Title level inconsistent:

Statistical Distributions
^^^^^^^^^^^^^^^^^^^^^^^^^
release/1.7.0-notes.rst:247: WARNING: Title level inconsistent:

Other
^^^^^
release/1.7.0-notes.rst:255: WARNING: Title level inconsistent:

New `scipy.stats.qmc` module
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
Hmm... I used `^` because that's supposed to be one level lower than `-` according to [this guide](https://documentation-style-guide-sphinx.readthedocs.io/en/latest/style-guide.html), and we want these nested within the improvements to `scipy.stats`. Currently it's rendering them in the ToC as separate sections:

![image](https://user-images.githubusercontent.com/6570539/119605910-7203bb00-bda6-11eb-978b-460a93430aa4.png)

Actually, the nesting is not what we want throughout - deprecated features is supposed to be a higher heading level than `scipy.linalg` deprecations, but it's rendering as a subsection of the QMC submodule. It looks to me like the `====` underlines is supposed to be a higher level heading than `----`, but it's rendering them as lower level heading. Is our heading system different from what the [documentation](https://documentation-style-guide-sphinx.readthedocs.io/en/latest/style-guide.html) suggests?

Or is it because we skipped H2 (over and underline with `*`). I can't render the documentation locally, but I wonder if it would fix things if everything that is currently H3, e.g.
```
New features
============
```
were promoted to H2
```
************
New features
************
```
everything that is H4 is promoted to H3, etc.